File: src/eu/pulsation/alarmsandbox/AlarmSandboxService.scala
package eu.pulsation.alarmsandbox

import android.widget.Toast
import android.app.Service
import android.os.{Binder, IBinder, Parcel, RemoteException}
import android.content.Intent
import android.util.Log

class AlarmSandboxService extends Service {

  override def onCreate() {
    Log.v("AlarmSandbox", "=> Service created")
    Toast.makeText(this, "Service created",
                   Toast.LENGTH_SHORT).show()
  }

  override def onBind(intent: Intent): IBinder = {
    mBinder
  }

  /**
   * This is the object that receives interactions from clients.
   * See RemoteService for a more complete example.
   */
  private final val mBinder: IBinder = new Binder {
    @throws(classOf[RemoteException])
    override protected def onTransact(code: Int, data: Parcel,
                                      reply: Parcel, flags: Int): Boolean = {
      super.onTransact(code, data, reply, flags)
    }
  }

}

File: src/eu/pulsation/alarmsandbox/AlarmSandboxService.java
// TODO: import code from http://stackoverflow.com/questions/4459058/alarm-manager-example

package eu.pulsation.alarmsandbox;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.IBinder;

public class AlarmSandboxService extends Service
{
    AlarmSandbox alarm = new AlarmSandbox();
    public void onCreate()
    {
        super.onCreate();       
    }

    public void onStart(Context context,Intent intent, int startId)
    {
        alarm.SetAlarm(context);
    }

    @Override
    public IBinder onBind(Intent intent) 
    {
        return null;
    }
    
}
